modules/reindex/src/main/java/org/elasticsearch/reindex/ReindexMetrics.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.reindex;
+
+import org.elasticsearch.telemetry.metric.LongHistogram;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+public class ReindexMetrics {
+
+    public static final String TOOK_TIME_HISTOGRAM = "es.reindex.took_time.histogram";
+
+    private final LongHistogram reindexTimeSecsHistogram;
+
+    public ReindexMetrics(MeterRegistry meterRegistry) {
+        this(meterRegistry.registerLongHistogram(TOOK_TIME_HISTOGRAM, "Time to reindex by search", "seconds"));
```

Comment:
are we sure that we want to use seconds granularity? we might end up reporting 0 seconds for fast reindex operations

Reply:
The max span of APM histograms is 100,000. That was the reason for choosing seconds granularity as per the discussion here: https://elastic.slack.com/archives/C05ENM6MEDD/p1722522943486389?thread_ts=1722522639.141469&cid=C05ENM6MEDD
Perhaps we can capture in centiseconds ? We probably don't expect operations to take more than 15 min ?

Reply:
Ah right, I didn't remember that limitation. Let's keep it in seconds then 👍